app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -32,30 +48,50 @@
 
 public class MainActivity extends AppCompatActivity {
 
+    private static final String COUNTER_TAG = "counter";
     private DrawerLayout mDrawerLayout;
     private Toolbar mToolbar;
     private NavigationView navigationView;
+    private ProgressBar downloadProgress;
+    private FrameLayout mainFrame;
+    private int counter;
+    private int eventsDone;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
-
-        DataDownload download = new DataDownload();
-        download.downloadVersions();
-
+        eventsDone = 0;
         setUpToolbar();
         setUpNavDrawer();
+        mainFrame = (FrameLayout) findViewById(R.id.layout_main);
         navigationView = (NavigationView) findViewById(R.id.nav_view);
+        downloadProgress = (ProgressBar) findViewById(R.id.progress);
+        downloadProgress.setVisibility(View.VISIBLE);
+        downloadProgress.setIndeterminate(true);
+        OpenEventApp.getEventBus().register(this);
+        DataDownload download = new DataDownload();
+        download.downloadVersions();
 
         this.findViewById(android.R.id.content).setBackgroundColor(Color.LTGRAY);
+
         FragmentManager fragmentManager = getSupportFragmentManager();
         fragmentManager.beginTransaction()
                 .replace(R.id.content_frame, new TracksFragment()).commit();
 
     }
 
+    @Override
+    protected void onStop() {
+        super.onStop();
+        OpenEventApp.getEventBus().unregister(this);
+    }
+
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        OpenEventApp.getEventBus().register(this);
```

Comment:
Why register is called twice? Once in onCreate and one time here, you can safely use onPause() and onResume() for Activity atleast, fragments is upto you.